javalib/src/main/scala/java/net/PlainSocketImpl.scala

```diff
@@ -389,13 +389,17 @@ private[net] class PlainSocketImpl extends SocketImpl {
     val bytesNum = socket
       .recv(fd.fd, buffer.asInstanceOf[ByteArray].at(offset), count.toUInt, 0)
       .toInt
-    if (bytesNum <= 0) {
-      if (errno.errno == EAGAIN || errno.errno == EWOULDBLOCK) {
+
+    bytesNum match {
+      case _ if (bytesNum > 0) => bytesNum
+
+      case _ if (bytesNum == 0) => if (count == 0) 0 else -1
```

Comment:
```suggestion
      case 0 => if (count == 0) 0 else -1
```

Reply:
@lolgab Thank you for the simplification.  I read this after this PR had been merged, so I created
PR #2149.  This code will be around for a long time and no sense letting an eyesore endure.